### PR TITLE
use createMock() instead of deprecated getMock()

### PR DESCRIPTION
**Pull request by OskarStark**
I am targeting this branch, because this is BC.

### CODE REVIEW

**Comment by OskarStark** [comment]
@greg0ire please review the latest commit

**Comment by jordisala1991** [comment]
I tried to do that on other bundle. He is not missing the class. He is trying to use this class from SonataAdminBundle. I couldn't make it work.

**Comment by greg0ire** [comment]
Hmm OK sorry... If it's not autoloadable it won't work.

**Comment by OskarStark** [comment]
Damn copy paste fail, my new class had the namespace `Sonata\AdminBundle` instead of ``Sonata\DoctrineORMAdminBundle`....

should be fixxed now

**Comment by greg0ire** [comment]
Please fix StyleCI @OskarStark

File: Tests/Datagrid/ProxyQueryTest.php [hunk 1]
```diff
@@ -45,46 +46,36 @@ public function dataGetFixedQueryBuilder()
...
+        $em = $this->createMock('Doctrine\ORM\EntityManager');
         $em->expects($this->any())
             ->method('getMetadataFactory')
             ->willReturn($mf);
         $em->expects($this->any())
             ->method('getConnection')
             ->willReturn($conn);
 
-        $q = $this->getMock('PDOStatement');
+        $q = $this->createMock('PDOStatement');
```
Comment (OskarStark) [inline Tests/Datagrid/ProxyQueryTest.php]: I think the error comes from this line here

```
1) Sonata\DoctrineORMAdminBundle\Tests\Datagrid\ProxyQueryTest::testGetFixedQueryBuilder with data set #0 ('aaa', 'bbb', 'id', 'id_idx', 33, 'integer')
PHPUnit_Framework_MockObject_RuntimeException: You cannot serialize or unserialize PDOStatement instances
/home/travis/build/sonata-project/SonataDoctrineORMAdminBundle/Tests/Helpers/PHPUnit_Framework_TestCase.php:56
/home/travis/build/sonata-project/SonataDoctrineORMAdminBundle/Tests/Datagrid/ProxyQueryTest.php:78
```

I found this, what do you think guys? And why is it failing now and not before?
https://github.com/sebastianbergmann/phpunit/issues/1171#issuecomment-37096058

Comment (greg0ire) [inline Tests/Datagrid/ProxyQueryTest.php]: Not sure if we need process isolation, maybe turn it off?

Comment (OskarStark) [inline Tests/Datagrid/ProxyQueryTest.php]: Its off by default, which should be the problem, according to the comment

Comment (jordisala1991) [inline Tests/Datagrid/ProxyQueryTest.php]: Maybe we can create an stdClass mock, with the required methods?

Not sure how to fix this otherwise :(

Comment (OskarStark) [inline Tests/Datagrid/ProxyQueryTest.php]: @jordisala1991 can you try to fix this and create a PR against mine?
Currently I don't have so much time... :(

Comment (greg0ire) [inline Tests/Datagrid/ProxyQueryTest.php]: @OskarStark @jordisala1991 can either of you fix this or just not change the test that fails? That way we can benefit from it in other PRs

Comment (jordisala1991) [inline Tests/Datagrid/ProxyQueryTest.php]: will take a look at it

Comment (jordisala1991) [inline Tests/Datagrid/ProxyQueryTest.php]: https://github.com/OskarStark/SonataDoctrineORMAdminBundle/pull/2

There you go. Not sure if it is the best solution, but on PHP < 5.6 you can't mock PDOStatement that easy

**Comment by OskarStark** [comment]
Thank you jordi

**Pull request merged**